realm/realm-library/src/main/cpp/io_realm_internal_SharedRealm.cpp

```diff
@@ -51,13 +54,16 @@ Java_io_realm_internal_SharedRealm_nativeCloseConfig(JNIEnv *, jclass, jlong con
 }
 
 JNIEXPORT jlong JNICALL
-Java_io_realm_internal_SharedRealm_nativeGetSharedRealm(JNIEnv *env, jclass, jlong config_ptr)
+Java_io_realm_internal_SharedRealm_nativeGetSharedRealm(JNIEnv *env, jclass, jlong config_ptr, jobject notifier)
 {
     TR_ENTER_PTR(config_ptr)
 
     auto config = reinterpret_cast<realm::Realm::Config*>(config_ptr);
     try {
         auto shared_realm = Realm::get_shared_realm(*config);
+        shared_realm->m_binding_context = JavaBindingContext::create(env, notifier);
```

Comment:
No, `JavaBindingContext` is holding a global weak ref there. And check the ref before using it. The life-cycle of `notifier` is totally controlled in java side -- as a member var of `SharedRealm.java`